diesel_migrations/migrations_internals/src/migration.rs

```diff
@@ -54,9 +54,15 @@ impl<'a> fmt::Display for MigrationFileName<'a> {
     }
 }
 
-pub fn migration_from(path: PathBuf) -> Result<Box<dyn Migration>, MigrationError> {
+pub fn migration_from<Conn>(
```

Comment:
As this is a function in our public API I would like to see some reasoning why this is change is required and what other possible solutions could be implemented.
(Basically this needs a reasoning why we need to break our API here)

Reply:
So, the explanation is that migration libraries would need to know the connection type when building the migration even before running it because they might want to do different things based on the type of database. This change provides that.

Alternative is that the migration libraries are compiled for just one database kind and then the migration builder and runner would just use logic for that connection kind. (This is what is happening with current barrel).

I discarded the alternative because I envision the future where production database might be different from test/dev database and we want to run migrations on both kind of connections with the same installations.

Reply:
Instead of passing in the whole connection I would abstract that away behind an `enum` as each backend could have more than one concrete connection type (for example the pooled variant, or a custom logger wrapper, or a future postgres connection based on rust-postgres, …)